### ISSUE TITLE
Vdebug interface is loaded also without the need for

### ISSUE THREAD

**Issue by przepompownia** [joonty/vdebug#295]
What is the meaning of loading `DebuggerInterface` or even whole `plugin/vdebug.vim` for vim sessions without files with type suitable to debugging by vdebug. 

Currently `plugin/vdebug.vim` is executed also at the start vim with an empty buffer. It should be necessary changed, both for the old and the new version, both for performance reasons and to avoid focusing on vdebug errors that are raised during open an empty buffers etc. 

Probably the solution is to move as most as possible code from `plugin` to `ftplugin` for each suitable file type and using common autoloaded functions to avoid code duplication.

**Comment by lucc** [comment]
I do not think that vdebug is a plugin that belongs into the `ftplugin` folder.  It can be used with different languages (and maybe support for even more can be added in the future?). Nevertheless you hit some important point there:

I was thinking about improving the startup of vdebug as well. But I found it difficult to implement the ideas I had ad hoc because of the layout of the code. Some changes are already merged in v2-integration (move to pythonx for example). But some other things are still difficult: For example how and where are vim `:command`s and `:map`s created. some from `plugin/vdebug.vim` but some also from within the python code. I hope to clean that up in order to have better control what is loaded and exposed when.

Maybe we can work together to find out what is needed. This would be my plan to go forward:
* make a list of actions that should be available (these should probably be implemented as autoload functions in order to improve loading time and can be bound to `:command`s or `:map`s by `plugin/vdebug.vim` and/or the python code)
  * global actions (that should be available everywhere):
    * start
    * stop
    * set breakpoint
    * ...
  * window local actions (for each window of the vdebug interface)
    * watch window
      * ...
    * log window
      * ...
    * ...
  * actions that only make sense when currently debugging (but should be available globally):
    * run to next breakpoint
    * step forward
    * step into
    * ...
* implement the autoload functions and then use them in all places where it makes sense
* if we have replaced the current interface with the new autoload functions in the backand (the names of maps and commands should probably stay the same) we can try to load and declare maps and commands lazyly wherever possible.

I have already started some of this work in my branch `interface` which is [here](https://github.com/lucc/vdebug/tree/interface).

**Comment by loonies** [comment]
I have 50+ plugins and Vdebug startup time is the slowest one - about 11x slower than the next to slowest. This is a huge slowdown, given Vdebug (at least in my case) is used rarely.

Using [hyiltiz/vim-plugins-profile](https://github.com/hyiltiz/vim-plugins-profile)

```
=====================================
Top 10 plugins slowing nvim's startup
=====================================
1       137.231   vim-vdebug
2        12.008   vim-nerdtree
3         5.032   vim-matchup
4         3.616   vim-delimate
...
=====================================
```
